src/core/ekat_rational_constant.hpp

```diff
@@ -152,13 +152,15 @@ struct RationalConstant {
 
   // These two are used to help reduce a/b to lowest terms
   static constexpr iType fix_num(const iType n, const iType d) {
-    return CONSTEXPR_ASSERT(d!=0),
-             n==0 ? n : n / gcd(abs(n),abs(d));
+    if (d==0)
```

Comment:
Throwing was not legal in C++11's constexpr functions, but it is ok now that we use C++17. So just do simple things here.